plugins/generator-1.20.1/forge-1.20.1/templates/block/block.java.ftl

```diff
@@ -345,10 +358,14 @@ public class ${name}Block extends
 
 	<#macro initCustomBlockStateProperties>
 		<#list data.customProperties as prop>
-			<#assign propName = prop.property().getName().replace("CUSTOM:", "")>
-			.setValue(${propName?upper_case},
+			<#assign propName = prop.property().getName()>
+			.setValue(${propName.replace("CUSTOM:", "")?upper_case},
 				<#if prop.property().getClass().getSimpleName().equals("StringType")>
-				${StringUtils.snakeToCamel(propName)}Property.${prop.value()?upper_case}
+					<#if propName.startsWith("CUSTOM:")>
+					${StringUtils.snakeToCamel(propName.replace("CUSTOM:", ""))}Property.${prop.value()?upper_case}
+					<#else>
+					${generator.map(propName, "blockstateproperties", 2)}.${blockpropvalues[propName][prop.value()]}

```

Comment:
Since you have reference to the `EnumProperty<T extends Enum<T>`, you can call EnumProperty#getValue to obtain this value.

This way blockpropertyvalues.json maintenance is not needed since this file is not needed.

Reply:
:heavy_check_mark: Fixed